src/mango.ts

```diff
@@ -0,0 +1,493 @@
+  hedgeDeltaTotal += dipDeltaDiff;
+  console.log(scalper.symbol, 'Adjust Slippage Delta by', dipDeltaDiff, 'to', -hedgeDeltaTotal);
+  const hedgeSide = IS_BUYSIDE ? HedgeSide.buy : HedgeSide.sell;
+
+  const notionalThreshold = deltaThreshold * fairValue;
+  const notionalAmount = -hedgeDeltaTotal * fairValue;
+  if ((notionalAmount < notionalThreshold && hedgeSide === HedgeSide.buy)
+       || (notionalAmount > notionalThreshold && hedgeSide === HedgeSide.sell)) {
+    console.log(scalper.symbol, 'Delta Netural: Slippage', deltaThreshold);
+    return;
+  }
+  console.log(scalper.symbol, 'Outside delta threshold:', Math.abs(hedgeDeltaTotal), 'vs.', deltaThreshold);
+
+  // Determine spot or perp order based on funding rate.
+  const bidSide = await perpMarket.loadBids(mangoClient);
+  const askSide = await perpMarket.loadAsks(mangoClient);
+  const fundingRate = (24 * 365)
+    * perpMarket.getCurrentFundingRate(bidSide, askSide);
+  const buySpot = fundingRate > PERP_FUNDING_RATE_THRESHOLD;
+  const sellSpot = -fundingRate < PERP_FUNDING_RATE_THRESHOLD;
+  let hedgeProduct = getMangoHedgeProduct(hedgeSide, buySpot, sellSpot);
+  // Do not allow Mango balance to go negative
+  if (mangoSpotDelta <= hedgeDeltaTotal) {
+    hedgeProduct = HedgeProduct.Perp;
+  }
+
+  console.log(
+    `${scalper.symbol} Target Delta Hedge: ${hedgeSide} ${hedgeProduct} ${-hedgeDeltaTotal} \
+      DIP Δ: ${dipTotalDelta} Mango Perp Δ: ${mangoPerpDelta} Mango Spot Δ: ${mangoSpotDelta} \
+      Spot Δ: ${spotDelta} Offset Δ ${scalper.deltaOffset} Fair Value: ${fairValue}`,
+  );
+
+  // Determine what price to use for hedging depending on allowable slippage.
+  hedgePrice = roundPriceToTickSize(IS_BUYSIDE
+    ? fairValue * (1 + slippageTolerance) : fairValue * (1 - slippageTolerance), scalper.tickSize);
+
+  // Break up order depending on whether the book can support it
+  const bookSide = IS_BUYSIDE ? askSide : bidSide;
+  const hedgeDeltaClip = roundQtyToSpotSize(hedgeDeltaTotal
+      / orderSpliceMango(
+        hedgeDeltaTotal,
+        fairValue,
+        maxNotional.get(scalper.symbol),
+        slippageTolerance,
+        bookSide,
+        perpMarket,
+      ), scalper.minSize);
+
+  const deltaOrderId = new Date().getTime() * 2;
+
+  // Start listening for Delta Hedge Fills
+  const deltaFillListener = async (event: WebSocket.MessageEvent) => {
+    const parsedEvent = JSON.parse(event.data as string);
+    const {
+      owner, clientOrderId, quantity, price,
+    } = parsedEvent.event;
+    if (owner === mangoAccount.publicKey.toBase58() && clientOrderId === deltaOrderId) {
+      const fillQty = (hedgeSide === HedgeSide.buy ? 1 : -1) * quantity;
+      hedgeDeltaTotal += fillQty;
+      console.log(`${scalper.symbol} Delta Filled ${hedgeSide} ${hedgeProduct} Qty ${fillQty} \
+          Price ${price} Remaining ${hedgeDeltaTotal} ID ${deltaOrderId} ${new Date().toUTCString()}`);
+    }
+  };
+
+  // Setup a listener for the order.
```

Comment:
If this logic is repeated in other places, should be refactored to be shared in a function like

addFillFeedIfSocketOpen

Reply:
TODO added